tools/test/examples/update.py

```diff
@@ -212,6 +212,12 @@ def upgrade_example(github, example, tag, user, ref):
                 os.chdir(cwd)
                 return False
 
+        checkout_branch = ['git', 'checkout', branch]
```

Comment:
force push does not sound right, I wonder why in the script it's ``-f`` argument. I would remove it ?

> We should have the option to create a branch if it doesn't already exist.

will add

> Also the option to force push directly to that branch on the upstream repo rather than creating a PR

Our flow is via pull requests. We should not directly push to upstream branches (no testing, no review?).